api/v1alpha2/linodemachine_types.go

```diff
@@ -66,8 +66,6 @@ type LinodeMachineSpec struct {
 	PrivateIP *bool `json:"privateIP,omitempty"`
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
 	Tags []string `json:"tags,omitempty"`
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
-	FirewallID int `json:"firewallID,omitempty"`
```

Comment:
do we want to delete this or deprecate it?

Reply:
I can add this back and FirewallRef's ID can take precedence if it's set

Reply:
I would be fine with firewallRef taking precedence if it is set over ID since we want to deprecate ID anyway. I think we also need a deprecation annotation on firewallID like
```
// +kubebuilder:deprecatedversionwarning=
```